src/js/paste.js

```diff
@@ -47,37 +47,53 @@ var PasteHandler;
     }
     /*jslint regexp: false*/
 
-    /* Paste Options:
-     *
-     * forcePlainText: Forces pasting as plain text. Default: true
-     * cleanPastedHTML: cleans pasted content from different sources, like google docs etc. Default: false
-     * cleanReplacements: custom pairs (2 element arrays) of RegExp and replacement text to use during paste when
-     *                    __forcePlainText__ or __cleanPastedHTML__ are `true` OR when calling `cleanPaste(text)`
-     *                    helper method. Default: []
-     * cleanAttrs: list of element attributes to remove during paste when __cleanPastedHTML__ is `true` or when
-     *             calling `cleanPaste(text)` or `pasteHTML(html,options)` helper methods.
-     *             Default: ['class', 'style', 'dir']
-     * cleanTags: list of element tag names to remove during paste when __cleanPastedHTML__ is `true` or when
-     *            calling `cleanPaste(text)` or `pasteHTML(html,options)` helper methods.
-     *            Default: ['meta']
-     *
-     * ----- internal options needed from base -----
-     * disableReturn
-     * targetBlank
-     * contentWindow
-     * ownerDocument
-     */
-
-    PasteHandler = function (instance, options) {
-        this.base = instance;
-        this.options = options;
-
-        if (this.options.forcePlainText || this.options.cleanPastedHTML) {
-            this.base.subscribe('editablePaste', this.handlePaste.bind(this));
-        }
-    };
+     PasteHandler = Extension.extend({
+
+        /* Paste Options */
+
+        /* forcePlainText: [boolean]
+         * Forces pasting as plain text.
+         */
+        forcePlainText: true,
+
+        /* cleanPastedHTML: [boolean]
+         * cleans pasted content from different sources, like google docs etc.
+         */
+        cleanPastedHTML: false,
+
+        /* cleanReplacements: [Array]
+         * custom pairs (2 element arrays) of RegExp and replacement text to use during paste when
+         * __forcePlainText__ or __cleanPastedHTML__ are `true` OR when calling `cleanPaste(text)` helper method.
+         */
+        cleanReplacements: [],
+
+        /* cleanAttrs:: [Array]
+         * list of element attributes to remove during paste when __cleanPastedHTML__ is `true` or when
+         * calling `cleanPaste(text)` or `pasteHTML(html, options)` helper methods.
+         */
+        cleanAttrs: ['class', 'style', 'dir'],
+
+        /* cleanTags: [Array]
+         * list of element tag names to remove during paste when __cleanPastedHTML__ is `true` or when
+         * calling `cleanPaste(text)` or `pasteHTML(html, options)` helper methods.
+         */
+        cleanTags: ['meta'],
+
+        /* ----- internal options needed from base ----- */
+        contentWindow: window,
+        ownerDocument: document,
```

Comment:
would suggest leaving these as this.base.options.ownerDocument internally ... but like the .document and .window idea. if they are in fact "core" options, that's where they should be read from